src/utilities/convex_hull.zig

```diff
@@ -291,7 +383,109 @@ fn rotateToXAxis(comptime Point: type, point: Point, angle: f64) ContinousPoint
     return ContinousPoint{ .time = newX, .value = newY };
 }
 
-test "incremental convex hull with known result" {
+/// Finds the tangent between convex hulls `hull_one` and `hull_two` for either the upper or lower
+/// hull determined by `hull_type`. This function ensures the resulting tangent line lies entirely
+/// outside both convex hulls and does not cross any points within the hulls. The function returns
+/// a struct containing the indices of the tangent points in `hull_one` and `hull_two`. It returns
+/// `IndexError` if either hull is empty.
+fn findTangent(
+    hull_one: *ArrayList(DiscretePoint),
+    hull_two: *ArrayList(DiscretePoint),
+    hull_type: HullType,
+) !struct {
+    hull_one_idx: usize,
+    hull_two_idx: usize,
+} {
+    // Ensure the input hulls are not empty.
+    if (hull_one.items.len == 0 or hull_two.items.len == 0) {
+        return Error.EmptyInput;
+    }
+
+    // Start with the rightmost point of hull_one and the leftmost point of hull_two.
+    var hull_one_idx: usize = hull_one.items.len - 1; // Rightmost point of hull_one.
+    var hull_two_idx: usize = 0; // Leftmost point of hull_two.
+
+    // Iterate until a valid tangent is found
+    while (true) {
+        // Check the turn direction at hull_two
+        const hull_two_turn = computeTurn(
+            hull_one.items[hull_one_idx], // Current point in hull_one
+            hull_two.items[hull_two_idx], // Current point in hull_two
+            hull_two.items[(hull_two_idx + 1) % hull_two.items.len], // Next point in hull_two
+        );
+
+        // Adjust hull_two_idx based on the turn and hull type
+        switch (hull_type) {
```

Comment:
Indeed, the conditions are different depending on the type of hull. So that simple line of code can only be executed if the condition on the `turn` is met. Another way to do that is with two conditions like:

if ((hull_type == HullType.upperHull and hull_two_turn == Turn.left) or (hull_type == HullType.lowerHull and hull_two_turn == Turn.right))

Should we change to that? I kind of like how it is now because it seems more clear and the comments help to make it also more clear. Not sure.

Reply:
I also think the current version is easy to read. However, it always looks like a mistake whenever a branching construct executes the same code for all branches. So maybe extend the comment right before the switch to make it clear that this intentional an explain why it is done?

Reply:
I think I found a better solution that considers your comment and keeps the switch statement for clarity.

Reply:
Thanks, I think the code is now more clear.